langchain-core/src/runnables/base.ts

```diff
@@ -12,11 +12,19 @@ import {
 } from "../tracers/log_stream.js";
 import { Serializable } from "../load/serializable.js";
 import { IterableReadableStream } from "../utils/stream.js";
-import { RunnableConfig, getCallbackMangerForConfig } from "./config.js";
+import {
+  RunnableConfig,
+  getCallbackMangerForConfig,
+  mergeConfigs,
+} from "./config.js";
 import { AsyncCaller } from "../utils/async_caller.js";
+import { Run } from "../tracers/base.js";
+import { RootListenersTracer } from "../tracers/root_listener.js";
 
 export type RunnableFunc<RunInput, RunOutput> = (
-  input: RunInput
+  input: RunInput,
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  options?: any
```

Comment:
We talked about making this

```
Record<string, any> & {
  config: RunnableConfig,
}
```

didn't we?

Reply:
Doing that caused type errors in other places where a function was being passed in and it had more than one arg. 
Those args were nullable so it didn't matter then, but now that we allow for a second arg if function's being passed don't match this type it errors.
Example: calling `runnable.pipe(formatDocumentsAsString)`
`formatDocumentsAsString` has two args: One of `Document` and a second optional arg of `string`.
If we type the second arg as something other than `any` it'll cause that to error.
Granted we could update those examples to this: `runnable.pipe((docs) => formatDocumentsAsString(docs))`
But anyone using this code would start to get type errors.